### ISSUE TITLE
Review comments on DAG stuff

### ISSUE THREAD

**Issue by danielhalm1407** [danielhalm1407/st310_project#23]
The comments:

Run the logit not just on categorical variables, but also on the integer ones. 

Plot the standard errors/confidence intervals in the coefficients of logistic model plot, so one can see which ones are statisticall significantly different from 0:

![Image](https://github.com/user-attachments/assets/afcfeea9-bebe-44f7-a009-7da30ec3a8ae)

Change interpretable to baseline:

![Image](https://github.com/user-attachments/assets/95c1f8d0-65a2-4baf-a05d-3cc09b8a79c2)

Good, I like how you show cp and oldpeak are significant.

Good, I like that loss function write-up

Get rid of the number as the top of the plot that are blocking the title misclassification rate vs lambda:

![Image](https://github.com/user-attachments/assets/2cb692f7-6271-4435-9a32-8c9e4a08651a)

Also, Ideally we should see a bell-curve so that an ideal value of lambda can be seen. We might need to cross validate a greater range of lambda. There should be a value of labda that is so large that the model becomes less accurate eventually. Also, lambda should be +ve, because it should increase the loss function. Please debug.

OK, selection of best two variable model appears good.

**DAG**

Please cite your mum or whatever other soucres you used. You can go on word and use the references function. Choose the source type as interview.

I like what you mention about diet -> put that in the DAG (maybe get rid of some others if you feel it becomes too complicated?) diet should have an arrow pointing to cholesterol and fasting blood sugar. But surely it affects other factors? Maybe this could be another source of confounding.

I like the mechanical causal chain that you go through. Would this mean that higher blood sugar only leads to target *through* the mechanism of higher blood cholesterol? if so, there shouldn't be an arrow directly from fbs to target. However, if it also has a direct effect, there should be an arrow.

Nice DAG

Explain more about confounding - say that if you were to use only trestbps in the logit, the log odds of target may appear to increase with an increase in trestbps more than they should do. This is because when trestbps rises, this is correlated with and hence coincides with an unobserved confounding increase in cholesterol, which also causes trestbps to rise. Thus, the logit coefficient (or perceived effect) of trestbps is biassed upward in magnitude, capturing some of the coincinding effect of the omitted variable of cholesterol (omitted variables bias, or OVB) and falsely attributing it to trestbps.

 I would first regress against trestbps only or chol only and explain the bias you just mentioned.

Your strategy of only regressing against fbs would work to eliminate OVB, but, in addition to this, another form of correcting for the bias is controlling for the confounders. So after you regress only against trestbps, add chol and fbs as controls. The coefficient then is how much does a unit rise in tresbps increase the log odds of having heart disease, holding cholesterol and blood sugar constant. Whenever trestbps changes independently of these two/ any variation in trestbps that is not correlated with chol and fbs is not confounded by them, and so the extent to which target is fit to depend on this variation in trestbps should be the extent to which it causally depends on it.

This ALSO applies to the other two regressors, chol and fbs. Thus, regressing target on trestbps, chol, and fbs, the coefficients can be interpreted as causal effects. So then you can say which factors increase your risk of heart disease the most. To interpet these better, standardise the regressors/predictors. This means the coefficients can be interpreted as the causal effect of a 1 standard deviation rise in their respective regressor (e.g., of a 1 sd rise in chol, what is the increase in the log odds of target).

Then, you could see which ones have the biggest coefficients and THEN say how a patient could fix this. Like if cholesterol has a really big impact on the log odds (i.e., relative chances) of having heart disease or not, you could then point to 'oh, but a low sugar diet is a big influencer of cholesterol levels through reducing blood sugar levels, and a low fat diet is a a big direct influencer of low cholesterol levels, so therefore, we would recommend a patient to make these changes to their diet!

Of course this last point about diet and cholesterol should be backed by citations of like a couple random studies online.

And then poom, that is the conclusion of our project/ us achieving our final key objective.
